### PR TITLE
add discordjs-typescript example

### PR DESCRIPTION
**Pull request by PantheraRed**
Added a separate discordjs-typescript example for typescript users.

### CODE REVIEW

File: examples/discordjs-typescript/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Client, Message } from 'discord.js';
+const client = new Client();
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`);
+});
+
+client.on('message', (msg: Message) => {
```
Comment (1chiSensei) [inline examples/discordjs-typescript/src/index.ts:8]:
```suggestion
client.on('message', (msg) => {
```
Is this type annotation needed?

File: examples/discordjs-typescript/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Client, Message } from 'discord.js';
```
Comment (1chiSensei) [inline examples/discordjs-typescript/src/index.ts:1]:
```suggestion
import { Client } from 'discord.js';
```
Is this type annotation needed?

**Review by 1chiSensei — CHANGES_REQUESTED** [review]

File: examples/discordjs-typescript/src/index.ts (hunk 2 above)
Comment (PantheraRed) [reply examples/discordjs-typescript/src/index.ts:1]: Yes, this is why I made a separate ts example. 🙂

**Comment by PantheraRed** [comment]
Sorry for mention @JakeCooper, but would you please review this pull request? 🙂

File: examples/discordjs-typescript/src/index.ts (hunk 1 above)
Comment (kakarot-dev) [reply examples/discordjs-typescript/src/index.ts:8]: That is what typescript is for

Comment (1chiSensei) [reply examples/discordjs-typescript/src/index.ts:8]: TypeScript auto detects it.

Comment (kakarot-dev) [reply examples/discordjs-typescript/src/index.ts:8]: that still doesn’t mean we dont have to give the annotation

**Review by kakarot-dev — COMMENTED** [review]
djsv13 will be coming out soon

**Comment by PantheraRed** [comment]
Yes & then normal discord.js example would require an update as well.

File: examples/discordjs-typescript/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const client = new Client();
```
Comment (kakarot-dev) [inline examples/discordjs-typescript/src/index.ts:2]: Specify some intents :)

File: examples/discordjs-typescript/src/index.ts (hunk 1 above)
Comment (kakarot-dev) [inline examples/discordjs-typescript/src/index.ts:8]: change from message to messageCreate

File: examples/discordjs-typescript/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`);
+});
+
+client.on('message', (msg: Message) => {
+  if (msg.content === 'ping') {
+    msg.reply('Pong 🏓');
+  } else if (msg.content === 'hello') {
+    msg.reply('Choo choo! 🚅');
```
Comment (kakarot-dev) [inline examples/discordjs-typescript/src/index.ts:12]:
```suggestion
    msg.reply({
    content: 'Choo choo! 🚅'
    });
```

File: examples/discordjs-typescript/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Client, Message } from 'discord.js';
+const client = new Client();
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`);
+});
+
+client.on('message', (msg: Message) => {
+  if (msg.content === 'ping') {
+    msg.reply('Pong 🏓');
```
Comment (kakarot-dev) [inline examples/discordjs-typescript/src/index.ts:10]:
```suggestion
    msg.reply({
    content: 'Pong 🏓'
    });
```

File: examples/discordjs-typescript/package.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  "main": "dist/index",
+  "author": "PantheraRed",
+  "license": "MIT",
+  "scripts": {
+    "dev": "nodemon src/index.ts",
+    "build": "tsc",
+    "start": "node ."
+  },
+  "dependencies": {
+    "discord.js": "^12.5.3"
```
Comment (kakarot-dev) [inline examples/discordjs-typescript/package.json:13]: Change djs to v13

**Review by kakarot-dev — CHANGES_REQUESTED** [review]
Djs v13

File: examples/discordjs-typescript/package.json (hunk 6 above)
Comment (PantheraRed) [reply examples/discordjs-typescript/package.json:13]: Not out yet but I'll make sure to commit those when v13 is officially out. :+1:

File: examples/discordjs-typescript/src/index.ts (hunk 5 above)
Comment (1chiSensei) [reply examples/discordjs-typescript/src/index.ts:10]: Isnt needed for v13.

File: examples/discordjs-typescript/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+title: Discord TypeScript bot
+description: A Discord bot written in TypeScript
+tags:
+  - discord.js
+  - typescript
+---
+
+# Discord.js Example
+
+This example starts a Discord bot using [discord.js](https://discord.js.org).
```
Comment (coffee-cup) [inline examples/discordjs-typescript/README.md:11]:
```suggestion
This example starts a Discord bot using [discord.js](https://discord.js.org) and [TypeScript](https://www.typescriptlang.org/).
```

**Review by coffee-cup — APPROVED** [review]
Thanks for this starter!

**Pull request merged**